core/src/main/scala/org/apache/spark/deploy/security/HadoopFSDelegationTokenProvider.scala

```diff
@@ -116,10 +117,10 @@ private[deploy] class HadoopFSDelegationTokenProvider
       if (fsToExclude.contains(fs.getUri.getHost)) {
         // YARN RM skips renewing token with empty renewer
         logInfo(s"getting token for: $fs with empty renewer to skip renewal")
-        fs.addDelegationTokens("", creds)
+        Utils.tryLogNonFatalError { fs.addDelegationTokens("", creds) }
```

Comment:
Looks okay as `fetchDelegationTokens` is actually wrapped in try/catch block that catches `NonFatal`. Previously it fails early and skips remaining tokens. This change makes it able to fetch more tokens if possible.

Just as there are still some tokens not available, the application is not guaranteed to be executed without issue like mentioned in the description:

> when the Spark app access nn-2 and nn-3, it will fail with o.a.h.security.AccessControlException: Client cannot authenticate via:[TOKEN, KERBEROS]

Although it is same as current behavior.

Reply:
Thanks for your review and summary, yes, with the above assumption that only `nn-2` has problems to issue token, accessing `nn-2` should fail even with this change, but the change guarantees that tokens are always available for `nn-1` and `nn-3`.

More context in our cases, only a few applications are going to access the unhealthy `nn-2`, but the current behavior causes a lot of jobs to fail.